src/components/vue-form-renderer.vue

```diff
@@ -205,42 +207,41 @@ export default {
         this.$emit('submit', this.transientData);
       }
     },
-    validateElements(elements) {
-      elements.forEach(element => {
-        if (element.validator && element.validator.errorCount !== 0) {
-          this.valid = false;
-          this.errors.push(element.validator.errors.errors);
+    getRules(items, data, rules) {
+      items.forEach(item => {
+
+        if (Array.isArray(item)) {
+          this.getRules(item, data, rules);
         }
-      });
-    },
-    validateContainer(container) {
-      if (container.$refs && container.$refs.container) {
-        this.validateContainer(container.$refs.container);
-      }
-      container.forEach(element => {
-        if (element.$refs && element.$refs.elements) {
-          this.validateElements(element.$refs.elements);
+
+        if (item.items) {
+          this.getRules(item.items, data, rules);
+        }
+
+        if (item.config && item.config.name && item.config.validation) {
+          data[`${item.config.name}`] = this.data[item.config.name];
+          rules[`${item.config.name}`] = item.config.validation;
         }
       });
+
     },
     isValid() {
       this.errors = [];
-      this.valid = true;
+      let data = {};
+      let rules = {};
+      this.getRules(this.config, data, rules);
```

Comment:
I'm not a fan of having function calls mutate the arguments it's applied to. It makes it hard to track the changes to these arguments, and hence makes it difficult to debug if there are issues.

I think having it return the new data and rules, like in the example below, will make this function easier to reason about, debug, and test:
```js
const { data, rules } = this.getRules(this.config);
```

And since it seems to return data _and_ rules, maybe it can be called `getDataAndRules`.

Reply:
fixed